include/repgp/repgp.h

```diff
@@ -59,155 +37,189 @@
 #include <stddef.h>
 #include <stdbool.h>
 #include <stdint.h>
-
 #include <json.h>
+
 #include "repgp_def.h"
-#include "errors.h"
-
-struct rnp_key_store_t;
-typedef struct pgp_packet_t    pgp_packet_t;
-typedef struct pgp_cbdata_t    pgp_cbdata_t;
-typedef struct pgp_stream_t    pgp_stream_t;
-typedef struct pgp_reader_t    pgp_reader_t;
-typedef struct pgp_cryptinfo_t pgp_cryptinfo_t;
-typedef struct pgp_io_t        pgp_io_t;
-typedef struct pgp_key_t       pgp_key_t;
-typedef struct pgp_pubkey_t    pgp_pubkey_t;
-
-/** pgp_region_t */
-typedef struct pgp_region_t {
-    struct pgp_region_t *parent;
-    unsigned             length;
-    unsigned             readc; /* length read */
-    unsigned             last_read;
-    /* length of last read, only valid in deepest child */
-    unsigned indeterminate : 1;
-} pgp_region_t;
-
-void pgp_init_subregion(pgp_region_t *, pgp_region_t *);
-
-/** pgp_cb_ret_t */
-typedef enum { PGP_RELEASE_MEMORY, PGP_KEEP_MEMORY, PGP_FINISHED } pgp_cb_ret_t;
-typedef pgp_cb_ret_t pgp_cbfunc_t(const pgp_packet_t *, pgp_cbdata_t *);
+
+#define REPGP_HANDLE_NULL ((void *) 0)
+
+typedef struct pgp_packet_t  pgp_packet_t;
+typedef struct pgp_stream_t  pgp_stream_t;
+typedef struct repgp_handle *repgp_handle_t;
+typedef struct repgp_io *    repgp_io_t;
+typedef uint32_t             rnp_result;
```

Comment:
I feel one more moving from `bool` to `rnp_result`, don't I? :)

Reply:
Indeed. I should include "rnp_def.h" instead redefining it here